test/topology_custom/test_mv_backlog.py

```diff
@@ -76,3 +81,112 @@ async def test_gossip_same_backlog(manager: ManagerClient) -> None:
+                         "AND tablets = {'initial': 1}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, v text, PRIMARY KEY (key, c))")
+    await cql.run_async(f"CREATE MATERIALIZED VIEW ks.mv_cf_view AS SELECT * FROM ks.tab "
+                    "WHERE c IS NOT NULL and key IS NOT NULL PRIMARY KEY (c, key) ")
+    await wait_for_view(cql, 'mv_cf_view', node_count)
+
+    # Only remote updates hold on to memory, so make the update remote
+    await pin_the_only_tablet(manager, "ks", "tab", servers[0])
+    await pin_the_only_tablet(manager, "ks", "mv_cf_view", servers[1])
+
+    # pin_the_only_tablet moves everything to shard 0
+    shard = 0
+
+    stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+
+    delay_metric_name = 'scylla_storage_proxy_coordinator_mv_flow_control_delay_total'
+    throttled_writes_metric_name = 'scylla_storage_proxy_coordinator_throttled_base_writes_total'
+
+    delay_limits = [0, 500, 1000, 2000, 10000]
+    computed_delays = []
+
+    local_metrics = await manager.metrics.query(servers[0].ip_addr)
+    previous_computed_delay = local_metrics.get(delay_metric_name, shard=str(shard)) or 0.0
+    previous_total_throttled_writes = local_metrics.get(throttled_writes_metric_name, shard=str(shard)) or 0.0
+
+    for delay_limit in delay_limits:
+        logger.info(f"delay_limit = {delay_limit}")
+        # Update the delay
+        await manager.server_update_config(servers[0].server_id, "view_flow_control_delay_limit_in_ms", delay_limit)
+
+        # Make sure that the next view update will hang
+        await manager.api.enable_injection(servers[0].ip_addr, "never_finish_remote_view_updates", one_shot=False)
+
+        def print_trace(res):
+            trace = res.get_query_trace()
+            logger.info("trace:")
+            for evt in trace.events:
+                logger.info(f"  {evt}")
+
+        # Generate a large view update
+        res = cql.execute(stmt, [0, 0, 100000*'a'], host=hosts[0], trace=False)
+        # print_trace(res)
+        # await asyncio.sleep(3)
+        # Generate a small write, to be slowed by the other view update
+        res = cql.execute(stmt, [0, 0, 'a'], host=hosts[0], trace=False)
+        # print_trace(res)
+
+        # Measure the current delay and make sure there was exactly one throttled write,
+        # or zero if we are currently testing 0ms delay
+        local_metrics = await manager.metrics.query(servers[0].ip_addr)
+        current_computed_delay = local_metrics.get(delay_metric_name, shard=str(shard)) or 0.0
+        current_total_throttled_writes = local_metrics.get(throttled_writes_metric_name, shard=str(shard)) or 0.0
+        if delay_limit == 0:
+            assert current_total_throttled_writes == previous_total_throttled_writes
+        else:
+            assert current_total_throttled_writes == previous_total_throttled_writes + 1
+
+        computed_delay = current_computed_delay - previous_computed_delay
```

Comment:
By the way, there is also a metric last_mv_flow_control_delay - maybe it will be easier to use than the one that measures the total?

Reply:
Yes, using `last_mv_flow_control_delay` here should result in a bit less code. I absent-mindedly switched from it to `throttled_base_writes_total` when I introduced the `mv_flow_control_delay_total` check.

Reply:
Actually, I tried it and it the test started to fail. IIRC the calculated last delay was zero. I didn't dig into why this happened, but my theory is that there might have been some other writes happening which do not have the backpressure applied (different replica set?) and might have reset this counter to zero.

I think that using `view_flow_control_delay_limit_in_ms` will be more reliable here anyway.